src/menu/create_wallet_menu.c

```diff
@@ -0,0 +1,145 @@
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "create_wallet_menu.h"
+
+#include "constant_texts.h"
+#include "ui_menu.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  GENERATE_NEW_WALLET,
+  RESTORE_FROM_SEED,
+} create_wallet_options_e;
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static const evt_config_t create_wallet_evt_config = {
+    .abort_disabled = false,
+    .evt_selection.byte = EVT_CONFIG_UI_EVT,
+    .timeout = MAX_INACTIVITY_TIMEOUT};
+
+const flow_step_t create_wallet_step = {
+    .step_init_cb = create_wallet_menu_initialize,
+    .p0_cb = NULL,
+    .ui_cb = create_wallet_menu_handler,
+    .usb_cb = NULL,
+    .nfc_cb = NULL,
+    .evt_cfg_ptr = &create_wallet_evt_config,
+    .flow_data_ptr = NULL};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+void create_wallet_menu_initialize(const engine_ctx_t *ctx,
+                                   const void *data_ptr) {
+  /* Create an array of pointers holding the string to display on the menu. */
+  char *menu_option_ptr_array[NUMBER_OF_OPTIONS_NEW_WALLET];
```

Comment:
make this declaration with `const ` so the rest of the typecasting can be removed without compiler warnings.

Reply:
fixed